src/main/java/org/takes/rs/RsWithHeader.java

```diff
@@ -94,22 +95,61 @@ public RsWithHeader(final Response res, final CharSequence name,
      */
     public RsWithHeader(final Response res, final CharSequence header) {
         super(
-            new Response() {
-                @Override
-                public List<String> head() throws IOException {
-                    final List<String> head = new LinkedList<String>();
-                    for (final String hdr : res.head()) {
-                        head.add(hdr);
-                    }
-                    head.add(header.toString());
-                    return head;
-                }
-                @Override
-                public InputStream body() throws IOException {
-                    return res.body();
-                }
-            }
+            new RsWithHeader.ResponseImpl(res, header)
+        );
+    }
+
+    private static final class ResponseImpl implements Response {
+        /**
+         * Pattern for all other lines in the head.
+         */
+        private static final Pattern HEADER = Pattern.compile(
+            "[a-zA-Z\\-]+:\\p{Print}+"
         );
+
+        /**
+         * Original response.
+         */
+        private final transient Response response;
+
+        /**
+         * Header to add.
+         */
+        private final transient CharSequence value;
+
+        /**
+         * Ctor.
+         * @param res Original response
+         * @param header Header to add
+         */
+        ResponseImpl(final Response res, final CharSequence header) {
+            if (!HEADER.matcher(header).matches()) {
+                throw new IllegalArgumentException(
```

Comment:
@YamStranger according to http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#constructors
> Without exceptions, there should be only one constructor that stores data in object variables

What do you think about it? F.e. `RsWithStatus` throws `IllegalArgumentException` in method `head`.